torch/csrc/distributed/rpc/init.cpp

```diff
@@ -561,11 +542,27 @@ If the future completes with an error, an exception is thrown.
          std::string& pickledPythonUDF,
          std::vector<torch::Tensor>& tensors,
          const float rpcTimeoutSeconds) {
-        DCHECK(!PyGILState_Check());
-        return pyRpcPythonUdf(
-            dst, pickledPythonUDF, tensors, rpcTimeoutSeconds);
+        return std::make_shared<jit::PythonFutureWrapper>(
+            pyRpcPythonUdf(dst, pickledPythonUDF, tensors, rpcTimeoutSeconds),
+            [](const py::object& value) {
+              py::gil_scoped_release release;
+              auto& pythonRpcHandler = PythonRpcHandler::getInstance();
+              // This will unwrap RemoteException and raise the contained
+              // server-side Python exception on client side. A caveat here is
+              // that the exception must be raise in the client thread calling
+              // the pybind "wait" API, so that it can be correctly shown to
+              // user. A wrong way is to raise it in RPC server thread, where
+              // the exception would be swallowed in the ThreadPool task, and
+              // also no pybind handling code can help shown the Python
+              // exception.
+              pythonRpcHandler.handleException(value);
+            } /* unwrap_func */);
       },
-      py::call_guard<py::gil_scoped_release>());
+      py::call_guard<py::gil_scoped_release>(),
+      py::arg("dst"),
+      py::arg("pickledPythonUDF"),
+      py::arg("tensors"),
+      py::arg("rf") = nullptr);
```

Comment:
this does not match the arg list in the lambda?

Reply:
Good catch. It's a rebasing-introduced bad piece. Removing.

It seems we have removed these arg list.
It also does not fail tests, so maybe it's only used by help(..).